File: python/백준/집합과 맵/1620 나는야 포켓몬 마스터 이다솜.py
import sys

input = sys.stdin.readline

N, M = map(int, input().split())


pocketMon_name = {}
pocketMon_int = {}
for i in range(1, N + 1):
    name = input().rstrip()
    pocketMon_name[name] = i
    pocketMon_int[i] = name

for i in range(M):
    quiry = input().rstrip()
    
    if quiry.isdigit():
        print(pocketMon_int[int(quiry)])
    else:
        print(pocketMon_name[quiry])
        
        
        

File: cpp/백준/집합과 맵/1620 나는야 포켓몬 마스터 이다솜.cpp
#include <iostream>
#include <string>
#include <vector>
#include <unordered_map>
#include <cctype>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    int N, M;
    cin >> N >> M;

    unordered_map<string, int> pocketMon_name;
    vector<string> pocketMon_int(N + 1);

    for (int i = 1; i <= N; ++i) {
        string name;
        cin >> name;
        pocketMon_name[name] = i;
        pocketMon_int[i] = name;
    }

    for (int i = 0; i < M; ++i) {
        string quiry;
        cin >> quiry;
        bool isDigit = !quiry.empty();
        for (char c : quiry) {
            if (!isdigit((unsigned char)c)) { isDigit = false; break; }
        }
        if (isDigit) {
            cout << pocketMon_int[stoi(quiry)] << "\n";
        } else {
            cout << pocketMon_name[quiry] << "\n";
        }
    }

    return 0;
}
